src/code_805A120.c

```diff
@@ -0,0 +1,1474 @@
+      flag = sub_8055640(pokemon,target,move,gUnknown_80F4F64,param_4) != 0 ? TRUE : FALSE;
+      sub_8079764(pokemon);
+  }
+  else {
+      SetChargeStatusTarget(pokemon,pokemon,STATUS_DIVING,move,*gUnknown_80FAD2C);
+      flag = TRUE;
+  }
+  return flag;
+}
+
+bool8 sub_80598CC(struct Entity *pokemon, struct Entity *target, struct Move *move, u32 param_4)
+{
+  bool8 flag;
+  
+  flag = FALSE;
+  if (sub_8055640(pokemon,target,move,0x100,param_4) != 0) {
+    flag = TRUE;
+    if(sub_805727C(pokemon, target, gUnknown_80F4E00))
+    {
+        LowerAccuracyStageTarget(pokemon, target, gUnknown_8106A4C, FALSE);
+    }
+  }
+  return flag;
+}
+
+bool8 StockpileMoveAction(struct Entity *pokemon, struct Entity *target, struct Move *move, u32 param_4)
+{
+    HandleStockpile(pokemon, target);
+    return TRUE;
+}
+
+bool8 sub_8059928(struct Entity * pokemon,struct Entity * target,struct Move * move,u32 param_4)
+{
+  bool8 flag;
+  s32 iVar2;
+  
+  iVar2 = 1;
+  flag = FALSE;
+  if ((u8)(target->info->chargingStatus - 7) <= 1){
+      iVar2 = 2;
+  }
+  if (sub_8055640(pokemon,target,move,iVar2 << 8,param_4) != 0)
+  {
+    flag = TRUE;
+    if(sub_805727C(pokemon,target,gUnknown_80F4DEC) != 0) {
+        CringeStatusTarget(pokemon,target,FALSE);
+    } 
+  }
+  return flag;
+}
+
+bool8 sub_8059988(struct Entity * pokemon,struct Entity * target,struct Move * move,u32 param_4)
+{
+  bool8 flag;
+  
+  flag = FALSE;
+  if (sub_8055640(pokemon,target,move,0x100,param_4) != 0) {
+    flag = TRUE;
+    gUnknown_202F218 = 1;
+  }
+  if (((gUnknown_202F218 != 0) && (gUnknown_202F214 == 2)) &&
+     (sub_805727C(pokemon,target,gUnknown_80F4DC8) != 0)) {
+    PoisonedStatusTarget(pokemon,target,FALSE);
+  }
+  return flag;
+}
+
+bool8 sub_80599EC(struct Entity *pokemon, struct Entity *target, struct Move *move, u32 param_4)
+{
+    s32 HP;
+
+    HP = target->info->maxHPStat;
+    if(HP < 0)
+        HP += 3;
+    HealTargetHP(pokemon, target, HP >> 2, 0, TRUE);
+    return TRUE;
+}
+
+bool8 sub_8059A18(struct Entity *pokemon, struct Entity *target, struct Move *move, u32 param_4)
+{
+    RaiseAccuracyStageTarget(pokemon, target, gUnknown_8106A50);
+    return TRUE;
+}
+
+bool8 sub_8059A2C(struct Entity * pokemon,struct Entity * target,struct Move * move,u32 param_4)
+{
+    u8 local_20;
+    u32 level;
+    
+    local_20 = 0;
+    level = pokemon->info->level;
+    sub_806F370(pokemon,target,level,1,&local_20,GetMoveType(move),sub_8057600(move,param_4),0,1,0);
+    local_20 = local_20 == 0;
+    return local_20;
+}
+
+bool8 sub_8059AA8(struct Entity *pokemon, struct Entity *target, struct Move *move, u32 param_4)
+{
+    ConfuseStatusTarget(pokemon, target, TRUE);
+    return TRUE;
+}
+
+bool8 sub_8059AB8(struct Entity *pokemon, struct Entity *target, struct Move *move, u32 param_4)
+{
+    TauntStatusTarget(pokemon, target);
+    return TRUE;
+}
+
+bool8 sub_8059AC4(struct Entity *pokemon, struct Entity *target, struct Move *move, u32 param_4)
```

Comment:
This is for the weather-based healing moves: Moonlight, Morning Sun, Synthesis.

Reply:
Are all three of them calling this function? I'm only seeing one call to this function.

Reply:
Chances are that they all use the same call, since those moves are identical besides the name. Likewise for Recover.

Reply:
Okay, I'll name it to 'WeatherHealMoveAction'?